pinot-core/src/main/java/com/linkedin/pinot/core/operator/filter/BitmapBasedFilterOperator.java

```diff
@@ -42,6 +42,8 @@
 
   public BitmapBasedFilterOperator(PredicateEvaluator predicateEvaluator, DataSource dataSource, int startDocId,
       int endDocId) {
+    // NOTE: predicate that is always false or true should not be passed into the ScanBasedFilterOperator
```

Comment:
Same concern as above.

Reply:
We can handle it, but we should not pass them in for performance concern.